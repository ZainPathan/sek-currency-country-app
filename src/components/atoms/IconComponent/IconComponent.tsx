import React from 'react';
import Icon from "@material-ui/core/Icon";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
    flag: {
        width: 80,
        height: 50
    }
});

interface IconComponentProps {
    src: string
}

const IconComponent: React.FC<IconComponentProps> = ({src}) => {
    const classes = useStyles();

    return (
        <Icon >
            <img alt='' className={classes.flag} src={src}/>
        </Icon>
    )
};

export default IconComponent;