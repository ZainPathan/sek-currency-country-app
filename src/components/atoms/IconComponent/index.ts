import IconComponent from './IconComponent';

export {
    IconComponent
};

export default IconComponent;