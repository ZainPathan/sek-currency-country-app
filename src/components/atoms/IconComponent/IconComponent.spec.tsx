import React from 'react';
import renderer from 'react-test-renderer';
import IconComponent from "./IconComponent";

describe('IconComponent tests', () => {

    const flagURL = 'https://restcountries.eu/data/bel.svg';

    it('snapshot test', () => {
        const tree = renderer
            .create(
                <IconComponent
                    src={flagURL}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})