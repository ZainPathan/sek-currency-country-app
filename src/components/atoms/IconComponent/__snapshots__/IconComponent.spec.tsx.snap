// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent tests snapshot test 1`] = `
<span
  aria-hidden={true}
  className="material-icons MuiIcon-root"
>
  <img
    alt=""
    className="makeStyles-flag-1"
    src="https://restcountries.eu/data/bel.svg"
  />
</span>
`;
